Add best practices section for rule 108 (#799)
zalando/restful-api-guidelines: chapters/compatibility.adoc

** Follow the redirect when the server returns HTTP status code {301} (Moved
  Permanently).

> The <<appendix-best-practices>> section contains recommendations on how to
> satisfy these requirements (<<client-handling-compatible-extensions>>).

Reviewer: ```suggestion
The <<handling-compatible-extensions>> section describes a best practice to implement the requirements in Java. 
```
